Source/Scene/PickDepth.js

```diff
@@ -0,0 +1,169 @@
+/*global define*/
+define([
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/PixelFormat',
+        '../Renderer/PixelDatatype',
+        '../Shaders/PostProcessFilters/PassThrough'
+    ], function(
+        defined,
+        defineProperties,
+        destroyObject,
+        PixelFormat,
+        PixelDatatype,
+        PassThrough) {
+    "use strict";
+    /*global WebGLRenderingContext*/
+
+    /**
+     * @private
+     */
+    var PickDepth = function(context) {
+        this.framebuffer = undefined;
+
+        this._depthTexture = undefined;
+        this._textureToCopy = undefined;
+        this._copyDepthCommand = undefined;
+
+        var supported = updateFramebuffers(this, context);
+        if (supported) {
+            updateCopyCommands(this, context);
+        }
+
+        this._supported = supported;
+    };
+
+    defineProperties(PickDepth.prototype, {
+        supported : {
+            get : function() {
+                return this._supported;
+            }
+        }
+    });
+
+    function destroyTextures(pickDepth) {
+        pickDepth._depthTexture = pickDepth._depthTexture && !pickDepth._depthTexture.isDestroyed() && pickDepth._depthTexture.destroy();
+    }
+
+    function destroyFramebuffers(pickDepth) {
+        pickDepth.framebuffer = pickDepth.framebuffer && !pickDepth.framebuffer.isDestroyed() && pickDepth.framebuffer.destroy();
+    }
+
+    function createTextures(pickDepth, context, width, height) {
+        pickDepth._depthTexture = context.createTexture2D({
+            width : width,
+            height : height,
+            pixelFormat : PixelFormat.RGBA,
+            pixelDatatype : PixelDatatype.UNSIGNED_BYTE
+        });
+    }
+
+    function createFramebuffers(pickDepth, context, width, height) {
+        destroyTextures(pickDepth);
+        destroyFramebuffers(pickDepth);
+
+        createTextures(pickDepth, context, width, height);
+
+        pickDepth.framebuffer = context.createFramebuffer({
+            colorTextures : [pickDepth._depthTexture],
+            destroyAttachments : false
+        });
+
+        if (pickDepth.framebuffer.status !== WebGLRenderingContext.FRAMEBUFFER_COMPLETE) {
```

Comment:
This was here because the color texture was float, but later I packed the depth into 3 unsigned bytes. According to the spec, a single RGBA/unsigned byte color attachment must be complete. I'll remove the check.